7_react/react1/week2/todo-app/src/timer/Timer.jsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react";
+
+const TimeCount = () => {
+  const [seconds, setSecond] = useState(0);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setSecond((prev) => prev + 1);
+    }, 1000);
+  });
```

Comment:
While technically valid, using useEffect without a dependency array might lead to unexpected behavior, especially in reusable components (maybe what you saw when trying to render it in thw TodoList?). Every time the component re-renders, the code within the useEffect hook will run. If TimeCount is used inside a component that triggers frequent re-renders, such as TodoList, the timer may behave unpredictably. To ensure consistent behavior and prevent external factors from affecting the timer, consider adding the 'seconds' variable to the dependency array. This ensures that the timer continues updating based solely on its own state, unaffected by external re-renders.
  `useEffect(() => {
    setTimeout(() => {
      setSecond((prev) => prev + 1);
    }, 1000);
  }, [seconds]);`